src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -55,31 +50,115 @@ double ThetaStarPathPlanner::coordDistance(const Coordinate &coord1,
     return distance(p1, p2);
 }
 
-bool ThetaStarPathPlanner::lineOfSight(const Coordinate &coord1, const Coordinate &coord2)
+bool ThetaStarPathPlanner::lineOfSight(const Coordinate &coord0, const Coordinate &coord1)
 {
-    CoordinatePair coord_pair(coord1, coord2);
-    // If we haven't checked this Coordinate pair for intersects before, check it now
-    auto line_of_sight_cache_it = line_of_sight_cache.find(coord_pair);
-    if (line_of_sight_cache_it == line_of_sight_cache.end())
+    int dy = coord1.col() - coord0.col();
+    int dx = coord1.row() - coord0.row();
+
+    if (std::abs(dy) < std::abs(dx))
     {
-        Segment seg(convertCoordToPoint(coord1), convertCoordToPoint(coord2));
-        bool has_line_of_sight = true;
+        if (coord0.row() > coord1.row())
+        {
+            return checkLineLow(coord1, coord0);
+        }
+        else
+        {
+            return checkLineLow(coord0, coord1);
+        }
+    }
+    else
+    {
+        if (coord0.col() > coord1.col())
+        {
+            return checkLineHigh(coord1, coord0);
+        }
+        else
+        {
+            return checkLineHigh(coord0, coord1);
+        }
+    }
+}
+
+bool ThetaStarPathPlanner::checkLineLow(const Coordinate &coord0,
+                                        const Coordinate &coord1)
```

Comment:
can checkLineLow and checkLineHigh be merged into one function with a flag? seems like a lot of duplicated code

Reply:
I have updated the implementation to combine the two functions using a flag.
I think the new implementation lowers the code readability, but I'm not sure if it's much different than before. Let me know if you think I should modify/keep the new version or revert to the original implementation.

Reply:
I think it's fine